### ISSUE TITLE
NET-1632 Fix S6444 FN: [GeneratedRegexAttribute] should require a timeout or RegexOptions.NonBacktracking

### ISSUE THREAD

**Issue by jamie-anderson-sonarsource** [SonarSource/sonar-dotnet#8520]
### Description

.NET 7 introduced two important updates to regexes:
* The `RegexOptions.NonBacktracking` flag for turning off backtracking in the engine.
* The `[GeneratedRegex]` attribute for compiling regexes during MSIL compilation rather than at runtime.

Rule S6444 is raised when a `Regex` object or method call potentially allows for a denial of service due to backtracking. This issue can be mitigated either by specifying an evaluation timeout or by passing the new `NonBacktracking` option.

S6444 is not currently applied to `[GeneratedRegex]`. The same logic that applies to `Regex` should apply to `[GeneratedRegex]`.

While the examples below are written in C#, the rule detection should equally apply to VB.NET.

### Example code

```csharp
using System.Text.RegularExpressions;

partial class Example
{
  [GeneratedRegex("^([A-Za-z]+)+$")] // Noncompliant (FN)
  public static partial Regex Noncompliant1();

  [GeneratedRegex("^([A-Za-z]+)+$", RegexOptions.None)] // Noncompliant (FN)
  public static partial Regex Noncompliant2();

  [GeneratedRegex("^([A-Za-z]+)+$", RegexOptions.NonBacktracking)] // Compliant
  public static partial Regex CompliantNonBacktracking();

  [GeneratedRegex("^([A-Za-z]+)+$", RegexOptions.None, 5000)] // Compliant
  public static partial Regex CompliantWithTimeout();
}
```

### Expected behavior

When `[GeneratedRegex]` is used, but `RegexOptions.NonBacktracking` is not used and no execution timeout is specified, S6444 should be raised.

### Actual behavior

S6444 is not raised as expected.

### Source

[APPSEC-243](https://sonarsource.atlassian.net/browse/APPSEC-243)

[APPSEC-243]: https://sonarsource-sandbox-608.atlassian.net/browse/APPSEC-243?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1632](https://sonarsource.atlassian.net/browse/NET-1632)

[NET-1632]: https://sonarsource.atlassian.net/browse/NET-1632?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by pavel-mikula-sonarsource** [comment]
Moved
